DB/src/main/java/io/deephaven/db/v2/locations/parquet/topage/ToDBDateTimePage.java

```diff
@@ -3,39 +3,77 @@
 import io.deephaven.db.tables.dbarrays.DbArray;
 import io.deephaven.db.tables.dbarrays.DbArrayDirect;
 import io.deephaven.db.tables.utils.DBDateTime;
+import io.deephaven.db.tables.utils.DBTimeUtils;
 import io.deephaven.db.v2.sources.chunk.Attributes;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
 import org.jetbrains.annotations.NotNull;
 
-public class ToDBDateTimePage<ATTR extends Attributes.Any> extends ToLongPage<ATTR> {
+import java.util.function.LongUnaryOperator;
 
-    private static ToDBDateTimePage INSTANCE = new ToDBDateTimePage<>();
+public abstract class ToDBDateTimePage<ATTR extends Attributes.Any> extends ToLongPage<ATTR> {
 
-    public static <ATTR extends Attributes.Any> ToDBDateTimePage<ATTR> create(@NotNull Class<?> nativeType) {
+    private static <ATTR extends Attributes.Any> ToDBDateTimePage<ATTR> fromUnitToNanos(final LongUnaryOperator convertUnitToNanos) {
+        return new ToDBDateTimePage<ATTR>() {
+            private long toNanos(final long v) {
+                return convertUnitToNanos.applyAsLong(v);
+            }
+
+            @Override
+            public final long[] convertResult(@NotNull final Object result) {
```

Comment:
This is definitely not the design we want. You don't need or want to override `convertResult` in the nanos case; array copy is the right approach in that scenario.